pkg/localmetrics/localmetrics.go

```diff
@@ -110,33 +111,43 @@ func NewMetricsCollector(store cache.Cache) *MetricsCollector {
 	}
 }
 
+type collectorMethod string
+
+const (
+	collectorDescribe collectorMethod = "Describe"
+	collectorCollect  collectorMethod = "Collect"
+)
+
+// runOnCollectors executes the named method from the Collector interface on each Collector
+// instance in the receiver, ignoring fields that don't implement Collector. The idea is to make
+// it easier to add a new metric; you don't have to remember to add it to our Describe and Collect
+// wrappers.
+// NOTE(efried): This is simplified based on knowing things about the Collector interface:
+// - There are only two methods, Describe and Collect, enforced here by the collectorMethod consts.
+// - Both methods take exactly one argument, so no need for `args` to be variadic.
+// - Neither method returns anything, so neither does this method.
+func (c *MetricsCollector) runOnCollectors(method collectorMethod, arg interface{}) {
+	// Start reflecting
+	cValue := reflect.ValueOf(*c)
+	// For each field in our MetricsCollector receiver...
+	for i := 0; i < cValue.NumField(); i++ {
+		// ...if the field is an instance of a Collector subclass...
+		if cValue.Type().Field(i).Type.Implements(reflect.TypeOf((*prometheus.Collector)(nil)).Elem()) {
+			// ...Find and invoke the indicated method.
+			cValue.Field(i).MethodByName(string(method)).Call([]reflect.Value{reflect.ValueOf(arg)})
```

Comment:
Bah, this doesn't work unless we export all the fields in MetricsCollector. Which wouldn't be the worst thing, but it's also not pretty.

/hold